crates/sparrow-main/tests/e2e/fixture/local_table.rs

```diff
@@ -82,11 +82,13 @@ impl LocalTestTable {
             //
             // TODO: Simulate the actual interaction with prepare (eg., collect raw files
             // and run prepare in response to analysis).
-            let mut iter = prepared_batches(source_data, &self.config, &None)
+            let mut prepared_iter = prepared_batches(source_data, &self.config, &None)
                 .await
                 .map_err(|e| e.into_error())?;
+            let mut iter = prepared_iter.stream();
```

Comment:
From a naming perspective, it's not clear why we *first* create an iterator *then* change it to a stream. It seems like either (a) we should create the stream directly, possibly with the iterator as a "hidden" implementation detail or (b) we should create the iterator and read from it directly.

Reply:
Hm, yeah. It was left as is since that's how prepare used it, but I can make a change to make this clearer now that we're using async streams.

Reply:
Okay made this change for the `prepare_iter` -> `prepare_input_stream`, let me know what you think and I'll do the same for the `execute_iter` in that case.

Reply:
That works for me.